### PR TITLE
[discover] Query editor UI changes

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description
* Move filter bar to a prepend filter button
* Add a series of query control: language reference, saved query management button, for PPL and SQL, add one more query control of toggle
* Build recent query pop over table, and able to run any recent query from history

## Screenshot

filter button: 

https://github.com/user-attachments/assets/d19124cc-c661-4ee8-a761-027751ae857f

query controls:
https://github.com/user-attachments/assets/18e1450f-4b3d-4dfb-9e8c-14495235cc6c

recent query:

https://github.com/user-attachments/assets/d82fc7b4-3bbd-4184-baf1-fc7e8c89fad6

## Changelog

- feat: Query editor UI changes

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/query/query_string/language_service/flyout_containers.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  flyoutHeader,
+  flyoutBody,
+  flyoutFooter,
+  ariaLabel,
+  size,
+}: Props) => {
+  return (
+    <div>
+      <EuiFlyout
+        className="observability-flyout"
```
Comment (kavilla) [inline src/plugins/data/public/query/query_string/language_service/flyout_containers.tsx:39]: ?

File: src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  const [isLanguageReferenceOpen, setIsLanguageReferenceOpen] = useState<boolean>(false);
+
+  const languageReferenceContainer = document.createElement('div');
+
+  const onCloseLanguageReference = () => {
+    ReactDOM.unmountComponentAtNode(languageReferenceContainer);
+    setIsLanguageReferenceOpen(false);
+  };
+
+  const osdDQLDocs = 'https://opensearch.org/docs/2.16/dashboards/dql)';
```
Comment (ashwin-pc) [inline src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx:52]: Can we get the version from package.json?

File: src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    id: 'languageReference',
+    label: i18n.translate('discover.queryControls.languageReference', {
+      defaultMessage: 'Open',
+    }),
+    testId: 'languageReference',
+    ariaLabel: i18n.translate('discover.queryControls.languageReference', {
+      defaultMessage: `Language Reference`,
+    }),
+    run: async (anchorElement) => {
+      if (props.queryLanguage === 'PPL' || props.queryLanguage === 'SQL') {
```
Comment (ashwin-pc) [inline src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx:70]: To keep the code cleaner, cant we register these when we register the PPL and SQL language using the additional controls property?

File: src/plugins/data/public/query/query_string/language_service/ppl_docs/commands/dedup.ts
Comment (ashwin-pc) [inline src/plugins/data/public/query/query_string/language_service/ppl_docs/commands/dedup.ts:1]: I'd move all these langugae docs to the query enhancements plugin

File: src/plugins/data/public/query/query_string/query_string_manager.ts [hunk 4]
```diff
@@ -112,9 +112,9 @@ export class QueryStringManager {
   };
 
   // Todo: update this function to use the Query object when it is udpated, Query object should include time range and dataset
```
Comment (ashwin-pc) [inline src/plugins/data/public/query/query_string/query_string_manager.ts:114]: Nit: We should remove this TODO

**Review by ashwin-pc — APPROVED** [review]
The styling and moving of the docs can be done in a fast follow. Approving the PR as is for now to unblock the rest of the work.

File: src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    }),
+    run: () => {
+      setIsCollapsed(!isCollapsed);
+      props.onToggleCollapse();
+    },
+    iconType: isCollapsed ? 'expand' : 'minimize',
+  };
+
+  const queryControls =
+    props.queryLanguage === 'PPL' || props.queryLanguage === 'SQL'
```
Comment (ashwin-pc) [inline src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx:150]: Lets not check explicitly for the PPL/SQL stuff here. The expand should be associated with the editor itself. If its the DQL editor, it should have this but if its the Default editor, then it should have this control alongside the other controls

**Review by kavilla — COMMENTED** [review]
Approving with agreement of fast follows.

Avoid hardcode ref to language in data plugin. Fast follow should consider modifying the contract of the language config to have more features. And if the language doesn't include it then dont render it. An example I would reference is Joshua's query editor extension. You can just render an array of actions.

For example you could pass in a button.

For the expand and collapsed I think it's weird the shifting of other buttons and that it's in the middle of the other actions even though it manipulates something not to the left of it. Like if I was tabbing through I could be typing and then I have to click tab twice otherwise the first button opens an overlay

Ive seen other applications include the expand and collapsed within the actual editor. Which makes sense for me because then I won't have  to misclick on the wrong button potentially and open an overlay if I wanted to go from the query editor to expand it

**Review by kavilla — APPROVED** [review]

**Pull request merged**